test/tools/integration/provision_master.sh

```diff
@@ -53,6 +53,11 @@ if ! ls kube-flannel.yml; then
   curl -LO https://raw.githubusercontent.com/coreos/flannel/v0.9.1/Documentation/kube-flannel.yml
   kubectl apply -f kube-flannel.yml
 fi
+if ! which go; then
```

Comment:
Just out of curiosity, why does it take so long, shouldn't it be just as parallelized as possible and t(e2eTest) == t(SlowestCreateAndDestroyMachine)?

Reply:
To be honest I don't know :) Haven't got much time to debug it. We could definitely follow that. I suspect it is a combination of too much concurrency and  the arbitrary time it takes to create/delete machine.

for example the following snippet shows that tests are paused and resumed, preemption ?
```
=== RUN   TestAWSProvisioningE2E/scenario_5_CoreOS_Docker_17.03
=== PAUSE TestAWSProvisioningE2E/scenario_5_CoreOS_Docker_17.03
=== CONT  TestAWSProvisioningE2E/scenario_1_Ubuntu_Docker_1.13
```

Reply:
do you know, how much `CPU, Memory` the machine we are running test on have ?

Reply:
Ah okay, I think we need to investigate that at some point, because IMHO running the tests scheduled at night instead of Pre/Post-merge is a bad idea, because there is no clear responsibility anymore in case they fail and 1h is definitely too long

Reply:
Or we start creating release branches and run the long test suite only when PRing into a release branch..

Reply:
Yeah, sounds good to me. I'd prefer to schedule a night test suite - this assures that the tests are executed frequently, at least at the beginning. The code base of the machine-controller doesn't grow very fast - a few commits during a day.

Reply:
@mrIncompetent just discussed about this and we would both like to have the full suite run on each PR and not scheduled. Also I think the runtime of this can and should definitely be improved.

If I understand this correctly, we currently only parralelize the machine scenarios, but not the cloudproviders?

E.G. we only start with DO scenarios once AWS finished? I think that should be changed.